hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpcengine/nio/SelectorOptimizer.java

```diff
@@ -163,6 +164,13 @@ public boolean remove(Object o) {
 
         @Override
         public boolean contains(Object o) {
+            if (!OS.isLinux()) {
```

Comment:
On Linux the functionality isn't called. I would just provide a correct implementation and leave out the isLinux parts (so what @arodionov said).

Reply:
Basically we only need this implementation to work in macos and windows. This is because:

1. We never saw this reproducable failure in linux before.
2. AFAIK, a single file descriptor can have 2 events in single poll in only non-linux os. In linux polled event is single and contains both flags (read and write is 5 for example).

For this reason, I didn't want to do this extremely frequent for loop in linux. Although most of the time it will loop only once, it will still do branch predictions etc.

TLDR; this is questionable but safe optimisation

Reply:
After talking with @pveentjer, we decided to remove this os specific logic 90d1b76.